pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -274,13 +274,15 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env model.Env
 				direction:        http_conn.INGRESS,
 			}
 			if listenerOpts.tlsMultiplexed {
+				// If multiplexed is needed, setup to filter chain match, one for "istio" ALPN TLS, one for default.
 				listenerOpts.filterChainOpts = []*filterChainOpts{
 					{
-						httpOpts:          httpOpts,
-						transportProtocol: authn.EnvoyRawBufferMatch,
+						httpOpts: httpOpts,
+						match: &listener.FilterChainMatch{
+							ApplicationProtocols: ALPNInMesh,
```

Comment:
pass the ALPN protocols as an option in filter chain opts. Don't build match here

Reply:
Either is fine to me. I just feel in future we can build filter chain match based on multiple fields, doing fancier filter chain match...
https://github.com/istio/istio/issues/6384#issuecomment-398259867